### ISSUE TITLE
[Bug] Args not effective in direct ModelSpec construction

### ISSUE THREAD

**Issue by xjing76** [matthewwardrop/formulaic#64]
```
import pandas as pd
from formulaic.model_spec import ModelSpec
import scipy

data = [[1,10],[2,12],[1,13]]
df = pd.DataFrame(data,columns=['a','b'])
formula = 'b~ 1+C(a)'
y_, X_ = formulaic.Formula(formula).get_model_matrix(df.head(1))

kwargs = {'output':'sparse'}
spec = ModelSpec(formula=formula, structure=X_.model_spec.structure,  **kwargs)
y, X = spec.get_model_matrix(df)
```

```
	Intercept	C(a)[T.2]
0	1.0	0
1	1.0	1
2	1.0	0

```
Here is a minimal example when we construct ModelSpec directly with additional args like `output ='sparse'` the return objects are still in dense format.

**Comment by matthewwardrop** [comment]
Hi @xjing76 ! Thanks for reporting.

There was a bug in formulaic 0.3.0 that would lead to this, but it should have been fixed in 0.3.1. Was it?

Your minimal example doesn't work for me (likely a different bug caused by the interaction of a structured formula with an unstructured model spec, looking into it now), but this code works fine against 0.3.1:

```
import pandas as pd
import formulaic
from formulaic.model_spec import ModelSpec
import scipy

data = [[1,10],[2,12],[1,13]]
df = pd.DataFrame(data,columns=['a','b'])
formula = 'b ~ 1+C(a)'
y, X_ = formulaic.Formula(formula).get_model_matrix(df.head(1))

kwargs = {'output':'sparse'}
spec = copy.deepcopy(X_.model_spec)
spec.output = 'sparse'
#spec = ModelSpec(formula=formula, structure=X_.model_spec.structure,  **kwargs)
X = spec.get_model_matrix(df)
X
```

```
<3x1 sparse matrix of type '<class 'numpy.int64'>'
	with 3 stored elements in Compressed Sparse Column format>
```

**Comment by matthewwardrop** [comment]
Currently, ModelSpec's spanning left/right hand sides are not supported. `ModelSpec`s only describe single `ModelMatrix` objects. There are ways around this, but they're not elegant... I'll try to think about this for future releases.

This bug, however, does seem to be resolved, so I will close this out. Feel free to reopen of 0.3.1 doesn't work for you. I'll create a separate issue for the improvement of the user experience re-direct `ModelSpec` creation.
